whois-freaks/operations.py

```diff
@@ -0,0 +1,112 @@
+""" Copyright start
+  Copyright (C) 2008 - 2023 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import validators, requests, json
+import base64
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('whois-freaks')
+
+
+class whoisFreaks(object):
+    def __init__(self, config, *args, **kwargs):
+        self.server_url = config.get('server_url')
+        if not self.server_url.startswith('https://'):
+            self.server_url = 'https://' + self.server_url
+        if not self.server_url.endswith('/'):
+            self.server_url += '/'
+        self.api_key = config.get('api_key')
+        self.sslVerify = config.get('verify')
+
+    def make_api_call(self, endpoint=None, method='GET', data=None, params=None):
+        try:
+            endpoint = self.server_url + endpoint
+            headers = {'Content-Type': 'application/json', 'Accept': 'application/json',
+                       "Authorization": "Token {}".format(self.api_key)}
+            response = requests.request(method, endpoint, params=params, headers=headers,
+                                        verify=self.sslVerify, data=json.dumps(data))
+            if response.ok:
```

Comment:
need to handle the response code 404 - Not found. In this case connector step could not fail.

Reply:
done